iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,275 @@
+## Plug and Play convention compatible APIs
+
+#### Common
+
+```csharp
+
+public abstract class PayloadConvention {
```

Comment:
Do you plan to add headers/summary in this PR? It'll definitely help with PR review

Reply:
I personally didn't want to do that here because of all of the clutter it would add and would spark more conversations about the docs than the API shape. 

I will add these comments when we're ready to get feedback on the docs.